fermipy/roi_model.py

```diff
@@ -2171,3 +2176,29 @@ def write_fits(self, fitsfile):
         hdu_data = fits.table_to_hdu(tab)
         hdus = [fits.PrimaryHDU(), hdu_data]
         fits_utils.write_hdus(hdus, fitsfile)
+
+    def write_ds9region(self, region, free='box',fixed='cross',frame='fk5',color='green',header=True):
+        """Create a ds9 compatible region file from the ROI
+        :param region: name of the region file (string)
+        :param free: one of the supported ds9 point symbols, used for free sources, see here: http://ds9.si.edu/doc/ref/region.html
+        :param fixed: as free, but for fixed sources
+        :param frame: typically fk5, more to be implemented
+        :param color: color used for the symbols
+        :param header: if True, will prepend a global header line.
+        :return: None.
+        """
+        allowed_symbols = ['circle','box','diamond','cross','x','arrow','boxcircle']
+        # adding some checks.
+        assert free in allowed_symbols, "symbol %s not supported"%free
+        assert fixed in allowed_symbols, "symbol %s not supported"%fixed
+        with open(region,'w') as fo:
+            if header:
+                fo.write("global color=%s\n"%color)
+            for src in self.get_sources():
+                if not isinstance(src,Source): continue
```

Comment:
when can this happen?

Reply:
not sure I understand this comment: 
header=False (as kwarg) will omit the first line

second, self.get_sources will return both Source, but also IsoSource and MapCube, in which case the sources should be ignored (since they are by construction all-sky and have no corresponding ds9 region string)

Reply:
Yeah, sorry, it was about `get_sources()`. I understand now. Maybe you could add this comment to the source code:

```
 self.get_sources will return both Source, but also IsoSource and MapCube, in which case the sources should be ignored (since they are by construction all-sky and have no corresponding ds9 region string)
```

Reply:
done as per latest commit.